### DISCUSSION TITLE
Unable to open “TSVD-inverse.srn5”network

### DISCUSSION THREAD

**Discussion by zhifeiji-han** [SCIInstitute/SCIRun#2282] [General] [answered]
When I try to open the example program“TSVD-inverse.srn5”  in SCIRun 5.0, the error is that legacy module not found. Is there a problem with the example itself ？

![image](https://user-images.githubusercontent.com/70640243/116021828-91120e80-a67b-11eb-9d44-657e7afb0514.png)

**Comment by dcwhite** [comment]
Looks like that module got renamed at some point (`SolveInverseProblemWithTikonovTSVD-->SolveInverseProblemWithTSVD`), and that network did not get updated in the toolkit. I'll get it fixed in the official version, but you can get the network to load right now by doing a simple text find/replace (the network file is just XML):

replace
`SolveInverseProblemWithTikhonovTSVD`
with 
`SolveInverseProblemWithTSVD`

everywhere in that file, and the network loads.

**Comment by zhifeiji-han** [comment]
Thank you！ According to your instructions, the program can load normally.

**Comment by zhifeiji-han** [comment]
I have a question. Is module SolveInverseProblemWithTikhonovSVD and module SolveInverseProblemWithTSVD functional the same? If not, what's the difference？

**Comment by jessdtate** [comment, marked as answer]
SolveInverseProblemWithTikhonovSVD and SolveInverseProblemWithTSVD are not the same.  TSVD is truncated SVD, where the matrix is condition by removing singular values in the solution.  TikhonovSVD uses Tikhonov regularization to condition the linear system.

**Comment by zhifeiji-han** [comment]
Thank you for the specific answer.
